hub/src/server/http.ts

```diff
@@ -74,7 +77,18 @@ export function makeHttpServer(config: HubConfigInterface): { app: express.Appli
         } else {
           writeResponse(res, { message: 'Server Error' }, 500)
         }
-      })
+      }
+    }
+
+    if (!asyncMutex.tryAcquire(endpoint, handleRequest)) {
+      const errMsg = `Concurrent operation (store) attempted on ${endpoint}`
+      logger.error(errMsg)
+      writeResponse(res, { 
+        message: errMsg, 
+        error: errors.ConflictError.name 
+      }, 409)
```

Comment:
Philosophical question -- should we be returning HTTP 409 or HTTP 429 here?  I remember saying 409 at one point since the client is (quite literally) making a conflicting request.  However, would it be better to advise the client on what to do here, instead of telling them what's wrong?  The advisable thing to do is to wait for a bit and try again, which would be indicated by HTTP 429.  Thoughts?

Reply:
I think HTTP 429 should be reserved for throttling or similar. The different error codes could inform the client on what retry/backoff/conflict-resolution strategy to attempt.